### PR TITLE
add expandMagclass helper

### PR DESCRIPTION
**Pull request by fbenke-pik**
Adds a helper to reshape read in magclass objects to match a reference so subsequent operations do not create problems due to mismatched dimensions.

See also https://github.com/pik-piam/remind2/issues/549

### CODE REVIEW

File: R/reportEmi.R [hunk 1]
```diff
@@ -143,8 +143,11 @@ reportEmi <- function(gdx, output = NULL, regionSubsetList = NULL,
   # FE non-energy use
   vm_demFENonEnergySector <- readGDX(gdx, "vm_demFENonEnergySector", field = "l",
                                      restore_zeros = F, react = "silent")[,t,]
+
   if (length(vm_demFENonEnergySector) == 0) {
     vm_demFENonEnergySector <- NULL
+  } else {
+    vm_demFENonEnergySector <- expandMagclass(vm_demFENonEnergySector, vm_demFeSector)
```
Comment (fbenke-pik) [inline R/reportEmi.R:150]: as this helper also removes values from third dimension that are not in `vm_demFeSector`, i am not sure if the trimming should be done here or only further down below in the context of the condition where it crashes (line 304)

Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [reply R/reportEmi.R:150]: Here is good.

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — APPROVED** [review]

**Review by orichters — APPROVED** [review]
thanks (probably you are still running buildLibrary?)

Maybe the proper place for this function would be in `magclass` somewhere, but I actually don't care.

**Comment by fbenke-pik** [comment]
> thanks (probably you are still running buildLibrary?)

I will, it just takes forever for remind2

**Pull request merged**